### PR TITLE
to #6957 implement interface deleteUpstreamRelation deleteDownstreamRelation 

### PR DESCRIPTION
**Pull request by zwZjut**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/6957
https://github.com/apache/dolphinscheduler/issues/6772

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 1]
```diff
@@ -114,29 +116,70 @@
...
     @Override
     public Map<String, Object> deleteUpstreamRelation(User loginUser, long projectCode, String preTaskCodes, long taskCode) {
-        return null;
+        Project project = projectMapper.queryByCode(projectCode);
+        //check user access for project
+        Map<String, Object> result = projectService.checkProjectAndAuth(loginUser, project, projectCode);
+        if (result.get(Constants.STATUS) != Status.SUCCESS) {
+            return result;
+        }
+        if (org.apache.commons.lang.StringUtils.isEmpty(preTaskCodes)) {
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: Please use DS's internal method to judge in controller.

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.java [hunk 2]
```diff
@@ -100,4 +101,42 @@ int deleteByCode(@Param("projectCode") long projectCode,
...
+
+
+    /**
+     * count upstream by codes
+     *
+     * @param projectCode projectCode
+     * @param taskCodes    taskCode
+     * @return upstream count list group by process definition code
+     */
+    List<Map<Long,Integer>> countUpstreamByCodeGroupByProcessDefinitionCode(@Param("projectCode") long projectCode, @Param("taskCodes")  long taskCodes);
```
Comment (brave-lee) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.java]: Please note the naming specification.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java [hunk 3]
```diff
@@ -240,4 +283,59 @@ public int hashCode() {
...
+                        processDefinitionCodeUpstreamCountMap.entrySet().stream().forEach(
+                                o -> {
+                                    Long processDefinitionCode = o.getKey();
+                                    Integer count = o.getValue();
+                                    List<ProcessTaskRelation> processTaskRelationList1 = processTaskRelationListGroupByProcessDefinitionCode.get(processDefinitionCode);
+                                    if (count <= processTaskRelationList1.size()) {
+                                        ProcessTaskRelation processTaskRelation = processTaskRelationList1.remove(0);
+                                        if (processTaskRelation.getPreTaskCode() != 0) {
+                                            processTaskRelation.setPreTaskCode(0);
+                                            updates.add(processTaskRelation);
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java:411]: Please set version=0

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java (hunk 1 above)
Comment (zwZjut) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessTaskRelationServiceImpl.java]: @RequestParam(name = "postTaskCodes", required = true) String postTaskCodes,  empty string is allowed

**Review by brave-lee — APPROVED** [review]
+1

**Pull request merged**